exercises/diamond/test/Tests.hs

```diff
@@ -3,15 +3,35 @@
 import Data.Foldable     (for_)
 import Test.Hspec        (Spec, describe, it, shouldBe)
 import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Test.QuickCheck   (Gen, elements, forAll, choose)
+import Data.List         ((\\))
+import Data.Maybe        (isNothing, fromMaybe)
+import Data.Char         (ord)
 import Diamond (diamond)
 
 main :: IO ()
 main = hspecWith defaultConfig {configFastFail = True} specs
 
 specs :: Spec
-specs = describe "diamond" $ for_ cases test
-  where
+specs = describe "diamond" $ do
+  it "non-Alpha characters should produce `Nothing`" $
+    forAll genNonAlphaChar $
+      isNothing . diamond
+
+  it "Length of a diamond should be odd" $
+    forAll genAlphaChar $ odd . length . fromMaybe [""] . diamond
+
+  it "Top and bottom of a diamond should be equal" $
+    forAll genAlphaChar $ topEqualToBottom . fromMaybe [""] . diamond
```

Comment:
This looks quite good.

I wonder how it might look like if we describe the actual property definition up here...

```haskell
  it "Top and bottom of a diamond should be equal" $
    forAllDiamond $ \dia ->
      let half = topLength dia
      in take half dia == take half (reverse dia)
```

Reply:
For the simpler PBTs, I think this style would be a bit less burdensome for future readers.

Reply:
Since we're using equality here, you can also consider [`===`](https://hackage.haskell.org/package/QuickCheck-2.13.2/docs/Test-QuickCheck.html#v:-61--61--61-). If it doesn't print the counter-example nicely, [`counterexample`](https://hackage.haskell.org/package/QuickCheck-2.13.2/docs/Test-QuickCheck.html#v:counterexample) can be used to pretty-print the upper and lower part for better visual feedback.